rowcontainer: add accounting for some memory use in DiskRowContainer

yuzefovich opened the pull request:
**sqlutils: remove option to print contents in CreateTable**

This was used only in a single test and I don't think we need this since it only pollutes the output.

Release note: None

**rowcontainer: make tests deterministic**

We were using global rand for shuffle and forgot to properly set randutil seed for tests.

Release note: None

**rowcontainer: add accounting for some memory use in DiskRowContainer**

This commit adds memory accounting for `scratchKey`, `scratchVal`, and keys stored in `deDupCache` if applicable in the disk row container in response to an OOM we saw during a roachtest run when we were spilling large geometry values to disk.

This required introducing a separate memory account bound to an unlimited memory monitor for this purpose. This is needed since the limited memory monitor we have in scope reached its limit, and that's why the row container had to spill to disk; however, we cannot spill the in-memory state of the disk row container, so it will only count against the root SQL memory monitor.

Fixes: #133337.

Release note: None

yuzefovich committed:
rowcontainer: make tests deterministic

We were using global rand for shuffle and forgot to properly set
randutil seed for tests.

Release note: None

yuzefovich committed:
rowcontainer: add accounting for some memory use in DiskRowContainer

This commit adds memory accounting for `scratchKey`, `scratchVal`, and
keys stored in `deDupCache` if applicable in the disk row container in
response to an OOM we saw during a roachtest run when we were spilling
large geometry values to disk.

This required introducing a separate memory account bound to an
unlimited memory monitor for this purpose. This is needed since the
limited memory monitor we have in scope reached its limit, and that's
why the row container had to spill to disk; however, we cannot spill the
in-memory state of the disk row container, so it will only count against
the root SQL memory monitor.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/136270)
<!-- Reviewable:end -->

yuzefovich committed:
sqlutils: remove option to print contents in CreateTable

This was used only in a single test and I don't think we need this since
it only pollutes the output.

Release note: None

DrewKimball reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 3 of 3 files at r1, 5 of 5 files at r2, 24 of 24 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/136270#-:-OD7voiZB0KehA2S1c9z:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @yuzefovich)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
TFTR!

bors r+
